### PR TITLE
doc(common): make authentication docs easier to find

### PR DESCRIPTION
**Pull request by coryan**
I created a page for all the authentication components. I also moved some general overview stuff out of the `Credentials` class into the components overview.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10110)

### CODE REVIEW

**Comment by coryan** [comment]
This is a rendered version, because the source may be hard to grok:

[pr-10110.pdf](https://github.com/googleapis/google-cloud-cpp/files/9882639/pr-10110.pdf)

**Review by scotthart — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10110)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @coryan)

File: google/cloud/doc/guac.dox [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/*!
+@defgroup guac Authentication Components
+
+Most services in Google Cloud Platform requires the client to authenticate the
+requests. Notable exceptions include public buckets in GCS and public data
+sets in BigQuery. The C++ client libraries are automatically configured
+to use "Google Default Credentials", some applications may need to override this
```
Comment (devbww) [inline google/cloud/doc/guac.dox:7]: s/some/but some/

File: google/cloud/doc/guac.dox [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+default. The functions and classes related to change the authentication
```
Comment (devbww) [inline google/cloud/doc/guac.dox:8]: s/change/changing/

File: google/cloud/doc/guac.dox [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+In most cases applications can control the [principal][principal-overview]
+used by the client libraries without having to change any code. By default the
+client libraries use [Application Default Credentials] which can be configured
+via environment variables, the `gcloud` CLI, or by changing the service account
+associated with your deployment environment (GCE, Cloud Run, GKE, etc.)
+
+## General Concepts
+
+As mentioned complete overview of authentication and authorization for Google
```
Comment (devbww) [inline google/cloud/doc/guac.dox:25]: s/mentioned /mentioned, /

File: google/cloud/doc/guac.dox [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Google Cloud Platform largely uses [OAuth2] access tokens for authentication.
+There are multiple ways to create such tokens. For example, when running on
+GCE the VM has access to a metadata server that can create these tokens for
+any application running on the VM. As another example, you can download a
+[service account keyfile] and the C++ client libraries will create access
+tokens using the contents of this file.
+
+Access tokens usually expire in about an hour. The client libraries
+automatically refresh these tokens when needed. The only exception is
+`MakeAccessTokenCredentials()` where the application provides the access token.
```
Comment (devbww) [inline google/cloud/doc/guac.dox:39]: s/ where/, where/

File: google/cloud/doc/guac.dox [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Development Workstations
+
+During development the most common configuration to use Application Default
```
Comment (devbww) [inline google/cloud/doc/guac.dox:43]: s/configuration to/configurations that/

File: google/cloud/doc/guac.dox [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+Credentials are:
+
+1. Use the `gcloud auth application-default` to authentication using the
```
Comment (devbww) [inline google/cloud/doc/guac.dox:46]: s/authentication/authenticate/

File: google/cloud/doc/guac.dox [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+   developer's account for authentication.
+1. Set `GOOGLE_APPLICATION_CREDENTIALS` environment variable to load a service
```
Comment (devbww) [inline google/cloud/doc/guac.dox:48]: s/Set/Set the/

File: google/cloud/doc/guac.dox [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+   account key. The value of this environment variable is the full path of a
+   file which contains the service account key.
```
Comment (devbww) [inline google/cloud/doc/guac.dox:50]: s/which/that/

Comment (coryan) [reply google/cloud/doc/guac.dox:50]: I think I fixed this and your other comments (thanks!) in #10111

**Pull request merged**